### ISSUE TITLE
Several issues with adal5-http.service

### ISSUE THREAD

**Issue by jo-me** [grumar/adal-angular5#3]
Hey there

I've integrated your adal wrapper and found several issues with the Adal5HTTPService.

1) Contrary to #1 the angular common http client's sendrequest has 3 arguments: method, url, options, see the debugger output here:
![image](https://user-images.githubusercontent.com/10351272/34732308-34b84df4-f565-11e7-8178-e1ea92597040.png)

If the method is omitted, then the request object becomes messed up. The URL is in the method field...
Not sure where the difference comes from, but I'm using Angular 5.1.

2) The authorization header is not added if there are already headers present in the request. 
adal-angular4 does not have this issue. 

Both issues can be fixed by replacing the block in sendRequest method with:

```
authenticatedCall = this.service.acquireToken(resource)
	.flatMap(function (token) {
		if (options.headers == null) {
			options.headers = new http_1.HttpHeaders();
		}
		options.headers = options.headers.append('Authorization', 'Bearer ' + token);
		return _this.http.request(method, url, options)
			.catch(_this.handleError);
	});
```

**Comment by grumar** [comment]
Fixed in d021fed3eae8fbacfb572280c307aebf1cdf69e1, please let me now if everything works like expected.

**Comment by geerzo** [comment]
This works for me. I would change 
`options.headers = options.headers.append('Authorization', 'Bearer ' + token);`
to
`options.headers = options.headers.set('Authorization', 'Bearer ' + token);`

This will ensure that if someone set that field already it's overwritten. Otherwise you wind up with 2 Authorization headers.

**Comment by pandianramalingam** [comment]
Hi @grumar ,

Set authorization header code throwing error if there are already headers present in the request.

BR,
PR

**Comment by geerzo** [comment]
Yes, the existing code has an error, this change fixes it.

**Comment by pandianramalingam** [comment]
when we can expect updated version ?

**Comment by grumar** [comment]
I made some changes in d021fed3eae8fbacfb572280c307aebf1cdf69e1  few days ago. Did You try lastest version?

**Comment by geerzo** [comment]
@grumar yes I have and it doesn’t work if the headers is already set because the local headers variable never gets set. You if block checks if options.headers is null then sets the headers variable but if that check fails the headers variable is never set but the next line sets the Authorization header and it fails because headers is null.

**Comment by jo-me** [comment]
The original issue is resolved.  THanks for that!

For more fault tolerance, I'd check if the options parameter is actually set because otherwise the request will fail altogether. If no special http options are needed the parameter could be omitted.

**Comment by stamler** [comment]
@jo-me Making observe: optional in the options object and using the HttpObserve type (like HttpClient does) should solve the problem. [I did it but haven't tested](https://github.com/stamler/adal-angular5/blob/master/src/adal5-http.service.ts) it yet. May be a nice update to @grumar 's  code.

**Comment by grumar** [comment]
Fixed in #22 & #23

**Comment by grumar** [comment]
published as beta
`npm install adal-angular5@2.1.1`
